src/binaryen-c.cpp

```diff
@@ -1605,6 +1605,14 @@ void BinaryenCallIndirectSetTarget(BinaryenExpressionRef expr,
   assert(targetExpr);
   static_cast<CallIndirect*>(expression)->target = (Expression*)targetExpr;
 }
+void BinaryenCallIndirectSetTableName(BinaryenExpressionRef expr,
+                                      const char* tableName) {
+  Name name(tableName);
+  auto* expression = (Expression*)expr;
+
+  assert(expression->is<CallIndirect>());
+  static_cast<CallIndirect*>(expression)->tableName = name;
+}
```

Comment:
It would be good to have `GetTableName` as a counterpart, as we have setter/getter pairs for other fields

Reply:
Thanks, I will add this. But in general I wasn't sure about the changes required to the C API. Like, we have a `BinaryenSetFunctionTable` which is based on the old assumption of having a single table. Also, should we have an overloaded `BinaryenCallIndirect` that takes the table name as well? I think we should.

Reply:
I think that changing the C API in whatever way makes sense is fine as long as the changes are documented in the changelog.